gnat2goto/driver/tree_walk.adb

```diff
@@ -3870,6 +3882,8 @@ package body Tree_Walk is
          Append_Argument (Call_Args, Param_Symbol (Ptr_Arg));
          Append_Argument (Call_Args, Param_Symbol (Len_Arg));
          Set_Arguments (Call_Inst, Call_Args);
+         Set_Lhs (I     => Call_Inst,
```

Comment:
Lhs_fun_call is set to the LHS of the returned duplicator function. Not sure why it needs that on top of the function return value (?) It's all a bit voodoo...

Reply:
I though about it a bit, and it sort of makes sense, i.e. to require function calls to have left-hand sides. Even if a function has a return value, if we didn't store it in the lhs, it would get lost.

Reply:
Hmm. You're saying as a temporary. I think this _could_ therefore be implicit in the make_code_function_call bla bla function.
The return value would be assigned at the call location - or the function can be called for its side effects (temporary not needed).
But nevermind. If that's how it is - that's how it is.

Reply:
I find it unfortunate that the relevant Irep does not require the lhs given that CBMC will fail. I'm not sure if make-function-call is the only place a function can be created, but yes: you are right.